specs/error_state/ErrorCodeStore.md

```diff
@@ -1,54 +1,40 @@
-`ErrorCodeStore` is for two original code store related errors: `CodeStoreOutOfGas` and 
-`MaxCodeSizeExceeded`.
-this type of error only occurs when executing create(create,create2) op code or tx deploy 
-transaction(tx.to = null).
+`ErrorCodeStore` is a combined error code and it's designed to handle code store related errors: `CodeStoreOutOfGas` and `MaxCodeSizeExceeded`. This type of error only occurs when executing `CREATE`/`CREATE2` opcode or a deployment transaction (tx.to = null).
 
 ### EVM behavior
-When handling a CREATE-kind transaction, Initial bytecode opcodes will run and current call context is created.
-The final bytecode opcodes to store for new contract is the `RETURN` opcode of init codes result.
+While handling a `CREATE`/`CREATE2` opcode, initial bytecode will be executed and then current call context is created. The contract bytecode will be returned through `RETURN` opcode as the execution result. `RETURN` returns a specified memory chunk [`offset`...`offset` + `length`] as bytecode. The gas cost for storing the bytecode is:
 
-`RETURN` opcode returns memory [`offset`...`offset` + `length`] content as bytecodes to store into state db.
-for returned bytecodes, store them cost additional gas.   
+```
+let CODE_DEPOSIT_BYTE_COST = 200
+code_store_cost = CODE_DEPOSIT_BYTE_COST * len(bytecodes)
+``` 
 
-`let CODE_DEPOSIT_BYTE_COST = 200
-code_store_cost = CODE_DEPOSIT_BYTE_COST * bytecodes' length
-`  
+- If `code_store_cost` > gas left, it is `CodeStoreOutOfGas` case.
+- If returned bytecode length > `MAXCODESIZE`, it is `MaxCodeSizeExceeded` case.  
 
-if `code_store_cost` > gas left, it is `CodeStoreOutOfGas` case.  
-if returned bytecodes' length > `MAXCODESIZE` allowed in evm, it is 
-`MaxCodeSizeExceeded` case.  
-
-In circuit bus mapping side, check these two code store errors in [here](https://github.com/privacy-scaling-explorations/zkevm-circuits/blob/main/bus-mapping/src/circuit_input_builder/input_state_ref.rs#L1148&L1155)
-when executing opcode is `RETURN` and call context is creating(`call.is_create == true`) meanwhile.  
-
-Even though errors occur in `CREATE` kind opcodes, it is special not checking error 
-in executing opcode `CREATE` directly.  
-Circuit implementation takes similar strategy, not constrain error directly in CREATE opcodes, but 
-in `RETURN` step context. Following this way it easy to get the key property state `length` and construct constraints against it.
+On the circuit bus mapping side, the checks for these two code store errors are [here](https://github.com/privacy-scaling-explorations/zkevm-circuits/blob/8a633f7c3de2da72f0817def57c1703241cced97/bus-mapping/src/circuit_input_builder/input_state_ref.rs#L1296-L1304). This error happens only when current opcode is `RETURN` and it's a `CREATE`/`CREATE2` call (`call.is_create == true`). We can't get contract bytecode length in `CREATE`/`CREATE2` opcodes and it's only available in `RETURN` opcode so we handle these two errors in `RETURN` opcode.
```

Comment:
```suggestion
On the circuit bus mapping side, the checks for these two code store errors are [here](https://github.com/privacy-scaling-explorations/zkevm-circuits/blob/8a633f7c3de2da72f0817def57c1703241cced97/bus-mapping/src/circuit_input_builder/input_state_ref.rs#L1296-L1304). This error only happens when both the current opcode is `RETURN` and the call is either `CREATE` or `CREATE2` call, that is when `call.is_create == true`. 

Even though these are `CREATE`/`CREATE2` errors, as we cannot get the contract bytecode length when processing these opcodes, we handle them in the `RETURN` opcode where the bytecode length is available.
```

Reply:
It might be worthwhile also adding a comment in the create.py file that the  the `CodeStoreOutOfGas` and `MaxCodeSizeExceeded`  errors are handled in the `RETURN` opcode instead.